ultravox/data/datasets.py

```diff
@@ -681,6 +683,51 @@ def _get_sample(self, idx, row) -> VoiceSample:
         return self._get_transcribe_sample(idx, row, tcol="text")
 
 
+class SodaDataset(VoiceDataset):
+    SYS_PROMPTS = [
+        "Follow the flow of the conversation and respond just like a human would in the same situation.",
+        "Engage in the conversation naturally, responding as a human would.",
+        "Follow the dialogue and reply like a person in that situation.",
+        "Participate in the chat and answer as if you were a human.",
+        "Interact smoothly and respond just like a person would.",
+        "Stay in the moment and reply as a human would in the conversation.",
+        "Flow with the discussion and respond naturally, as a person would.",
+        "Keep the dialogue going and answer like a human would.",
+        "Follow along and reply in a way a person would in the chat.",
+        "Stay engaged in the conversation and respond like a human.",
+        "Maintain the flow of the chat and answer just as a person would.",
+    ]
+
+    def __init__(self, args: VoiceDatasetArgs) -> None:
+        super().__init__(args)
+        dataset = self._load_audio_dataset(
+            "fixie-ai/soda-audio", split=args.split.value
+        )
+        self._init_dataset(dataset)
+
+    def _get_sample(self, idx, row) -> VoiceSample:
+        turns = row["dialogue"]
+        # Make sure the last turn is the assistant's
+        roles = ["user", "assistant"] if len(turns) % 2 == 0 else ["assistant", "user"]
+
+        num_prompts = min(self._args.num_prompts, len(self.SYS_PROMPTS))
+        sys_prompt = self.SYS_PROMPTS[idx % num_prompts]
```

Comment:
did we end up using a RNG for this sort of thing rather than the index?

Reply:
(I forget where but we discussed adding a private RNG to datasets to allow them to simply pull a value from the RNG rather than using the index counter and various moduli)

Reply:
Yes, the idea was that we do that in the next PR.

Reply:
Might as well just do it now I guess since I have the code.

Reply:
Done.